src/sql-parser/src/parser.rs

```diff
@@ -6032,115 +6032,179 @@ impl<'a> Parser<'a> {
         match self.parse_privilege_specification() {
-            Some(privileges) => {
-                self.expect_keyword(ON)?;
-                // If the object type is omitted, then it is assumed to be a table.
-                let object_type = self.parse_object_type().unwrap_or(ObjectType::Table);
-                match object_type {
-                    ObjectType::View | ObjectType::MaterializedView | ObjectType::Source => {
-                        return parser_err!(
+            Some(privileges) => self.parse_revoke_privilege(privileges),
+            None => self.parse_revoke_role(),
+        }
+    }
+
+    /// Parse a `REVOKE PRIVILEGE` statement, assuming that the `REVOKE` token
+    /// and all privileges have already been consumed.
+    fn parse_revoke_privilege(
+        &mut self,
+        privileges: PrivilegeSpecification,
+    ) -> Result<Statement<Raw>, ParserError> {
+        self.expect_keyword(ON)?;
+        let objects = if self.parse_keyword(ALL) {
+            let object_type = self.expect_grant_revoke_plural_object_type_in_schema("REVOKE")?;
+            self.expect_keywords(&[IN, SCHEMA])?;
+            let schemas = self.parse_comma_separated(Parser::parse_schema_name)?;
+            GrantObjectSpecification {
+                object_type,
+                object_spec_inner: GrantObjectSpecificationInner::All { schemas },
+            }
+        } else {
+            let object_type = self.expect_grant_revoke_object_type("REVOKE")?;
+            let names =
+                self.parse_comma_separated(|parser| parser.parse_object_name(object_type))?;
+            GrantObjectSpecification {
+                object_type,
+                object_spec_inner: GrantObjectSpecificationInner::Objects { names },
+            }
+        };
+        self.expect_keyword(FROM)?;
+        let roles = self.parse_comma_separated(Parser::expect_role_specification)?;
+        Ok(Statement::RevokePrivileges(RevokePrivilegesStatement {
+            privileges,
+            objects,
+            roles,
+        }))
+    }
+
+    /// Parse a `REVOKE ROLE` statement, assuming that the `REVOKE` token
+    /// has already been consumed.
+    fn parse_revoke_role(&mut self) -> Result<Statement<Raw>, ParserError> {
+        let role_name = self.parse_identifier()?;
+        self.expect_keyword(FROM)?;
+        let member_names = self.parse_comma_separated(Parser::expect_role_specification)?;
+        Ok(Statement::RevokeRole(RevokeRoleStatement {
+            role_name,
+            member_names,
+        }))
+    }
+
+    /// Bail out if the current token is not an object type suitable for a GRANT/REVOKE, or consume
+    /// and return it if it is.
+    fn expect_grant_revoke_object_type(
+        &mut self,
+        statement_type: &str,
+    ) -> Result<ObjectType, ParserError> {
+        // If the object type is omitted, then it is assumed to be a table.
+        let object_type = self.parse_object_type().unwrap_or(ObjectType::Table);
```

Comment:
Is this copying postgres? Could we instead make it not assumed but required?

Reply:
This does copy postgres. I'm a bit torn on this. If the parsed object type always matched the actual object type then I would prefer it to be required. However, when granting privileges on views, materialized views, and sources, you have to specify table. So for example let's say there's some view v, then `GRANT _ ON VIEW v TO _` is invalid, you must say `GRANT _ ON TABLE v TO _` or `GRANT _ ON v TO _`. So it is nice that you can omit table when the object isn't actually a table.

We could also allow `GRANT _ ON VIEW v TO _`, but we'd probably also want the other two version to be valid anyway for postgres compatibility.

Reply:
Oh I forgot about the silly tables are views rule. Ok, fine is as.